File: First_Palindrome/first.go
package firstpalindrome

func isPalindrome(s string) bool {
	for i, j := 0, len(s)-1; i < j; i, j = i+1, j-1 {
		if s[i] != s[j] {
			return false
		}
	}

	return true
}

func firstPalindrome(words []string) string {
	for _, word := range words {
		if isPalindrome(word) {
			return word
		}
	}
	return ""
}

File: First_Palindrome/first.js
/**
 * Check if a string is palindrome
 * @param {string} s - String to check
 * @returns {boolean} returns whether s is palindrome
 */
const isPalindrome = (s) => {
	let i = 0;
	let j = s.length - 1;

	while (i < j) {
		if (s[i] !== s[j]) {
			return false;
		}
		i++;
		j--;
	}

	return true;
};
/**
 * @param {string[]} words - The array of word
 * @returns {string} returns the first found palindrome if any
 */
var firstPalindrome = function (words) {
	for (let i = 0; i < words.length; i++) {
		if (isPalindrome(words[i])) {
			return words[i];
		}
	}

	return "";
};
